src/sections/upload-dataset-files/uploaded-files-list/UploadedFiles.tsx

```diff
@@ -0,0 +1,190 @@
+
+interface DatasetFilesProps {
+  fileUploadState: FileUploadState[]
+  cancelTitle: string
+  saveDisabled: boolean
+  updateFiles: (file: FileUploadState[]) => void
+  cleanup: () => void
+  addFiles: (fileUploadState: FileUploadState[]) => void
+}
+
+export function UploadedFiles({
+  fileUploadState,
+  cancelTitle,
+  saveDisabled,
+  updateFiles,
+  cleanup,
+  addFiles
+}: DatasetFilesProps) {
+  const { t } = useTranslation('uploadDatasetFiles')
+  const [selected, setSelected] = useState(new Set<FileUploadState>())
+  const [filesToRestrict, setFilesToRestrict] = useState<FileUploadState[]>([])
+  const [filesToAddTagsTo, setFilesToAddTagsTo] = useState<FileUploadState[]>([])
+  const [tagOptions, setTagOptions] = useState([
+    t('tags.documentation'),
+    t('tags.data'),
+    t('tags.code')
+  ])
+  const [terms, setTerms] = useState('')
+  const [requestAccess, setRequestAccess] = useState(true)
+  const [showRestrictionModal, setShowRestrictionModal] = useState(false)
+  const [showAddTagsModal, setShowAddTagsModal] = useState(false)
+  const updateFilesRestricted = (files: FileUploadState[], restricted: boolean) => {
+    if (restricted) {
+      setFilesToRestrict(files)
+      setShowRestrictionModal(true)
+    } else {
+      files.forEach((file) => (file.restricted = false))
+      updateFiles(files)
+    }
+  }
+  const restrict = (res: RestrictionModalResult) => {
+    if (res.saved) {
+      setTerms(res.terms)
+      setRequestAccess(res.requestAccess)
+      filesToRestrict.forEach((file) => (file.restricted = true))
+      updateFiles(filesToRestrict)
+    }
+    setShowRestrictionModal(false)
+  }
+  const addTags = (res: AddTagsModalResult) => {
+    if (res.saved) {
+      const files = filesToAddTagsTo.map((file) => {
+        res.tags.forEach((t) => {
+          if (!file.tags.includes(t)) file.tags.push(t)
+        })
+        return file
+      })
+      updateFiles(files)
+    }
+    setShowAddTagsModal(false)
+  }
+  const deleteFile = (file: FileUploadState) => {
+    file.removed = true
+    setSelected((x) => {
+      x.delete(file)
+      return x
+    })
+    updateFiles([file])
+  }
+  const updateSelected = (file: FileUploadState) => {
+    setSelected((current) => {
+      if (current.has(file)) {
+        current.delete(file)
+      } else {
+        current.add(file)
+      }
+      return new Set<FileUploadState>(current)
+    })
+  }
+  const save = () => {
+    addFiles(fileUploadState)
+    cleanup()
+  }
+
+  return (
+    <>
+      <div className={styles.forms}>
+        <RestrictionModal
+          defaultRequestAccess={requestAccess}
+          defaultTerms={terms}
+          show={showRestrictionModal}
+          update={restrict}
+        />
+        <AddTagsModal
+          show={showAddTagsModal}
+          availableTags={[...tagOptions]}
+          setTagOptions={setTagOptions}
+          update={addTags}
+        />
+      </div>
```

Comment:
Modals don't need to be wrapped by anything just inside the fragments its ok as they wont live inside this div when they are opened.